core-rust/core-api-server/src/core_api/browse.rs

```diff
@@ -63,67 +116,348 @@ impl<'s, S: SubstateDatabase> EngineStateMetaLoader<'s, S> {
+                enabled_features: vec![],
+                substituted_generic_types: vec![],
+            },
+        }))
+    }
+
+    /// An implementation delegate of [`Self::load_entity_meta()`] for `KeyValueStore`s.
+    fn load_kv_store_meta(
+        &self,
+        node_id: &NodeId,
+        kv_store_info: KeyValueStoreInfo,
+    ) -> Result<KeyValueStoreMeta, EngineStateBrowsingError> {
+        let KeyValueStoreGenericSubstitutions {
+            key_generic_substitution,
+            value_generic_substitution,
+            ..
+        } = kv_store_info.generic_substitutions;
+        let resolver = TypeReferenceResolver::new(&self.reader);
+        Ok(KeyValueStoreMeta {
+            resolved_key_type: resolver
+                .resolve_generic_substitution(Some(node_id), key_generic_substitution)
+                .and_then(|resolved_type| self.augment_with_schema(resolved_type))?,
+            resolved_value_type: resolver
+                .resolve_generic_substitution(Some(node_id), value_generic_substitution)
+                .and_then(|resolved_type| self.augment_with_schema(resolved_type))?,
+        })
+    }
+
+    /// Loads metadata of all fields and collections within the blueprint referenced by the given
+    /// [`BlueprintTypeTarget`]. The "type target" will also be used while resolving all types (see
+    /// [`TypeReferenceResolver`]).
+    fn load_blueprint_state_meta(
+        &self,
+        type_target: &BlueprintTypeTarget,
+    ) -> Result<ObjectModuleStateMeta, EngineStateBrowsingError> {
         let blueprint_id = &type_target.blueprint_info.blueprint_id;
-        let blueprint_definition =
-            self.reader
-                .get_blueprint_definition(blueprint_id)
-                .map_err(|error| {
-                    EngineStateBrowsingError::UnexpectedEngineError(
-                        error,
-                        "when getting blueprint definition".to_string(),
-                    )
-                })?;
+        let IndexedStateSchema {
+            fields,
+            collections,
+            ..
+        } = self
+            .reader
+            .get_blueprint_definition(blueprint_id)
+            .map_err(|error| {
+                EngineStateBrowsingError::UnexpectedEngineError(
+                    error,
+                    "when getting blueprint definition".to_string(),
+                )
+            })?
+            .interface
+            .state;
 
-        let fields = (0..blueprint_definition.interface.state.num_fields())
-            .map(|field_index| {
-                field_index
-                    .try_into()
-                    .expect("guaranteed by max field count")
-            })
-            .map(|field_index| {
+        let resolver = TypeReferenceResolver::new(&self.reader);
+
+        let fields = fields
+            .into_iter()
+            .flat_map(|(_partition_description, fields)| fields)
+            .enumerate()
+            .map(|(index, schema)| {
```

Comment:
_(note from author)_

I just realized that this logic does NOT filter out the fields disabled by features. It is not that trivial (e.g. needs another DB read to support "feature on outer" condition), but the fix is coming (maybe even in this PR).